packages/core/content-manager/server/bootstrap.js

```diff
@@ -6,4 +6,12 @@ module.exports = async () => {
   await getService('components').syncConfigurations();
   await getService('content-types').syncConfigurations();
   await getService('permission').registerPermissions();
+
+  // TODO: V5, remove this warning
+  if (strapi.config.get('server.webhooks.populateRelations', true)) {
```

Comment:
Why are we deprecating this, the ability to populate the relations in webhooks is an extremely common need and is one of the core reasons we have webhooks?

I really don't believe this message should exist and we should evaluate reworking webhooks in v5 to add more features not remove current ones.

Reply:
I explain it to derrick since I found it really hard to explain in text Darrick recorded a loom for me to explain it to you. the issue I see with not doing populateRelations for auditlogs
https://www.loom.com/share/b175f2b9ce5a4e41a0d36793a3a4682e

Reply:
you are completely right. I will remove this log and we will not deprecate the ability to populate relations in webhooks.

Two comments about this:
- Populating relations in webhooks greatly impacts performance of every CRUD operation of the Admin panel. I think in the future (v5) we should set this parameter to not populate relations by default. wdyt?
- Webhooks will only have (potentially) populated relations when creating the entity through the Admin panel. If you use the API and do not set the populate parameter, you will not have any relation in the webhook payload.

Reply:
@Marc-Roig For auditlogs you would not have to populate all relations I would exactly change it to only give the ids of the 
CRUD operation for content-type relations also to my opionion populating it in the entity manager would not be the most performant way Best way I see to do this so it only affects troughput is doing it on the eventHub level aka it would fire like ENTRY_CREATE_WITOUT_POPULATE and someting in strapi is subscripted to that and would fire ENTRY_CREATE 

1 This would only work poperly for content-type relations and not for components 
2 a issue could acure where the ENTRY_CREATE_WITOUT_POPULATE fires after somone else did somting else already like deleting an entery so then you would not be able to populate it anymore and can only show the id

Reply:
@Boegie19 wow that's a nice idea. It would not impact at all the API performance, and would bring consistency to how the webhooks behave.

It would have eventual consistency and could work for most of the user base. And for compatibility reasons, we could leave the ENTRY_CREATE and add this new ENTRY_CREATE_WITH_POPULATE.

I would say this would work for all operations ( create, update, publish, unpublish) except for delete. At the moment the event hub receives the event, the entity is already deleted.

So for that case we could populate the relation in the entityService itself.

Reply:
@Marc-Roig For delete it would also work for relations that are not content types if you give it the ids of them that would work for content type relations but not for components so components you would have to do in entity manager.

Reply:
Mmm, but at the ENTRY_CREATE_WITHOUT_POPULATE we woulnd't have ids , but the count of how many they are.
So I understand when trying to populate relations after the entry is deleted, we wouldn't have a way of knowing which relations the entry had.

Reply:
Also I would not mind making a pull request to write that only information I would need is where I would have to put the eventHub processor for ENTRY_CREATE_WITOUT_POPULATE to ENTRY_CREATE + What we want to still populate inside the entity manager. aka only componets for delete or all components for everyting and only do content types trough the webhook

Reply:
> @Boegie19 At the ENTRY_CREATE_WITHOUT_POPULATE hook we woulnd't have ids , but the count of how many they are. So I understand when trying to populate relations after the entry is deleted, we wouldn't have a way of knowing which relations the entry had.

Could we change the count to an id list?
Since that would make the doing it later possible 

Also components would have to be done inside the entity manager still since with every update you can delete components so then my question for auditlogs becomes should deletes be shown somehow.

Reply:
It could be possible to have populated ids, but could impact performance as close as populating the whole list  of attributes I am afraid. 
Some users have more than 200k relations in their entries and populating the list of ids could impact their performance too.

Also the Admin panel expects the count.

> Also components would have to be done inside the entity manager still since with every update you can delete components so then my question for auditlogs becomes should deletes be shown somehow.

The deletion of components is done in the entity manager if I recall correctly (maybe Im wrong, I would have to check)
And about audit logs 🤔 that's a good question. @derrickmehaffy wdyt?

Reply:
> Also I would not mind making a pull request to write that only information I would need is where I would have to put the eventHub processor for ENTRY_CREATE_WITOUT_POPULATE to ENTRY_CREATE + What we want to still populate inside the entity manager. aka only componets for delete or all components for everyting and only do content types trough the webhook

I will take a look where we can place that processor for the eventHub, I will see if there is something remotely similar to this.

Reply:
> It could be possible to have populated ids, but could impact performance as close as populating the whole list of attributes I am afraid.

It depends on the relationship In one to many where you are on the one side it does not have to look at any other tables for the many to many it only has to look at the middle table and not at the other end so it would improve performance it would still be bad but less bad in a one to many on the many side it would not make any difference expect that you send less data so it should take less time to send and receive

Reply:
Hellou :)
I think the issue with Audit Logs is a separate issue. Not populating in the webhooks could be handled by audit log itself (by populating on their side, or by inspecting the update query before the update etc.). Also Audit log may want to see how to handle many relations too, how do you do when you have a million relations ? Do you store all of them each time someone edit something on the entry?
Populating all relations by default is something that should never be done anywhere in Strapi. It has to be limited to X relations at a time, otherwise, the first time you have an entry with many relations, the audit logs and all the app may break.
How do you handle this case @derrickmehaffy?

Reply:
@petersg83 With the current implementation it is not it uses the eventHub event this fires.
I see 3 options
1 Do it anyway (bad)
2 Find a way to only show all changes (better) still would change how webhooks work so would be a breaking change
3 Don't merge this at all because of the performance hit.

Reply:
@remidej Just to confirm whatever we do to the event hub event. Aka how we fix this will also effect audit logs right? 

So my question then is is this the  separate issue from audit logs yes or no?

Reply:
to Boegie's point @petersg83 it's exactly that, Webhooks, Audit Logs, and DB Lifecycles all use the event hub. If you mess with one you mess with all of them so regardless they are all connected. The only to adjust one without the others is to separate them which defeats the purpose of the eventhub.

Personally I think it makes since the way it's implement (everything talk to the eventhub) but we haven't even decided if the eventhub is even stable ( I mean AFAIK we consider it experimental since it's literally not documented at all).